zephyr/module.yml

```diff
@@ -1,6 +1,8 @@
 # Copyright (c) 2021 Nordic Semiconductor ASA
 # SPDX-License-Identifier: Apache-2.0
 
+name: example-application
```

Comment:
Please suggest text describing this setting, covering:
How is this setting used (e.g., used to derive directory names?, injected into documentation text?)
What restrictions are there no the value (e.g., no spaces?, all lower-case?, only alphanumeric, underscore, hyphen?)